### ISSUE TITLE
Rule Request: AvoidUnusedAsignments

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#220]
Sonar rule -  java:S1854 - Unused assignments should be removed
leads to a lot of false positives which makes the rule unusable.
Create a rule which detects unused assignments.
example:

```
   String doodle = "init"; // bad, assigned value overwritten, not used
        doodle = "k";
        String str = "first"; // good
        String other = str.toString(); // used
        str = "empty"; // bad
        doSomething();
        str = "other"; // good
        doSomethingWith(str);
        str = ""; // bad
        str = "second";

```
